### PR TITLE
Update pixi configurations

### PR DESCRIPTION
**Pull request by travishathaway**
With the release of `0.13.0` pixi now supports defining multiple environments. This is beneficial for this package because it helps us more easily test different versions of Python. This should also make the CI test runners go quicker as they need to spend less time recalculating the lock file.

### CODE REVIEW

**Review by pavelzw — COMMENTED** [review]
btw you can also do `default = ["dev"]` which results in `pixi run test` using the dev feature directly.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -46,29 +46,24 @@ jobs:
...
         os: ["macos-latest", "ubuntu-latest", "windows-latest"]
 
     steps:
       - uses: actions/checkout@v3
 
       - uses: prefix-dev/setup-pixi@v0.3.0
         with:
-          pixi-version: v0.6.0
+          pixi-version: v0.17.0
           cache: true
```
Comment (pavelzw) [inline .github/workflows/tests.yml:58]:
```suggestion
          cache: true
          environments: ${{ matrix.test-env }}
```
image
https://pixi.sh/latest/advanced/github_actions/#multiple-environments

Comment (pavelzw) [reply .github/workflows/tests.yml:58]: also, `cache: true` is the default nowadays (you need to update setup-pixi to 0.5.1 anyway

**Comment by travishathaway** [comment]
Thanks for the recommendations @pavelzw.

**Pull request merged**